### PR TITLE
Update Access Report school_year column name

### PR DESCRIPTION
**Pull request by TurnerRiley**
UPDATE: This was going to require several more changes to account for the different `CensusSubmission` models that rely on the `school_year`.

The `CensusSummary` table's `school_year` column is actually confusing because it's really the Access Report year. This PR updates the column name for clarity.

## Links
Jira ticket: [here](https://codedotorg.atlassian.net/jira/software/c/projects/ACQ/boards/64?assignee=60d62161f65054006980bd71&assignee=5ff3bcc9a9d0300107c5456e&selectedIssue=ACQ-1628)

## Testing story
Locally ran `db:migrate` successfully.
SUCCESSFUL DB

### CODE REVIEW

**Review by hannahbergam — APPROVED** [review]
Love this! Founders mentality!

File: dashboard/db/migrate/20241025182954_update_census_summary_school_year_name.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class UpdateCensusSummarySchoolYearName < ActiveRecord::Migration[6.1]
+  def change
+    rename_column :census_summaries, :school_year, :access_report_year
```
Comment (drizco) [inline dashboard/db/migrate/20241025182954_update_census_summary_school_year_name.rb:3]: I'm curious if this also needs to drop the old index and apply the new one, or does ActiveRecord take care of this for us?

Comment (drizco) [reply dashboard/db/migrate/20241025182954_update_census_summary_school_year_name.rb:3]: here's an old migration that drops an old index and adds a new one https://github.com/code-dot-org/code-dot-org/blob/543bcc8d8ca04bd6996373cc3caddf3776fdf79a/dashboard/db/migrate/20131212190308_unmodify_followers_index.rb

Comment (TurnerRiley) [reply dashboard/db/migrate/20241025182954_update_census_summary_school_year_name.rb:3]: I originally did the drop/add but saw that without it, ActiveRecord [took care of it](https://github.com/code-dot-org/code-dot-org/pull/62030/files#diff-962f2f608ae3ac7ce84f1a84a6a43f4f14eada4d98fd5530c0b0905555a6ffafL15) for me!

Comment (drizco) [reply dashboard/db/migrate/20241025182954_update_census_summary_school_year_name.rb:3]: more rails magic!! 🪄 

<s>could you maybe add a screenshot? `show index from census_summaries;`</s>

Comment (drizco) [reply dashboard/db/migrate/20241025182954_update_census_summary_school_year_name.rb:3]: ooohhh, is `schema.rb` a generated file? I get it now 😁

**Review by drizco — APPROVED** [review]

File: dashboard/db/migrate/20241025182954_update_census_summary_school_year_name.rb (hunk 1 above)
Comment (TurnerRiley) [reply dashboard/db/migrate/20241025182954_update_census_summary_school_year_name.rb:3]: Yep yep!

**Pull request closed without merging**